apiserver/allfacades.go

```diff
@@ -263,9 +263,9 @@ func registerWatchers(registry *facade.Registry) {
 	registry.MustRegister("RelationStatusWatcher", 1, newRelationStatusWatcher, reflect.TypeOf((*srvRelationStatusWatcher)(nil)))
 	registry.MustRegister("RelationUnitsWatcher", 1, newRelationUnitsWatcher, reflect.TypeOf((*srvRelationUnitsWatcher)(nil)))
 	registry.MustRegister("RemoteRelationWatcher", 1, newRemoteRelationWatcher, reflect.TypeOf((*srvRemoteRelationWatcher)(nil)))
-	registry.MustRegister("VolumeAttachmentsWatcher", 2, newVolumeAttachmentsWatcher, reflect.TypeOf((*srvMachineStorageIdsWatcher)(nil)))
-	registry.MustRegister("VolumeAttachmentPlansWatcher", 1, newVolumeAttachmentPlansWatcher, reflect.TypeOf((*srvMachineStorageIdsWatcher)(nil)))
-	registry.MustRegister("FilesystemAttachmentsWatcher", 2, newFilesystemAttachmentsWatcher, reflect.TypeOf((*srvMachineStorageIdsWatcher)(nil)))
+	registry.MustRegister("VolumeAttachmentsWatcher", 2, newMachineStorageIdsWatcher, reflect.TypeOf((*srvMachineStorageIdsWatcher)(nil)))
+	registry.MustRegister("VolumeAttachmentPlansWatcher", 1, newMachineStorageIdsWatcher, reflect.TypeOf((*srvMachineStorageIdsWatcher)(nil)))
+	registry.MustRegister("FilesystemAttachmentsWatcher", 2, newMachineStorageIdsWatcher, reflect.TypeOf((*srvMachineStorageIdsWatcher)(nil)))
```

Comment:
Love the simplification. We can move these into the storageprovisioner facade if you want.

See https://github.com/juju/juju/blob/9781c2c0f6633e7f156771b94249e48b8090bc58/apiserver/facades/agent/migrationminion/register.go#L19 on how I did it for the migration status watcher. Only makes sense if it is only for one facade.

Reply:
Yeah, that's what I initially did(Because I thought they are controller facades). But I realized that these are agent facades. So we have to keep them like this.

Reply:
Like this new pattern.